src/generateSdk/generateSdkApi.ts

```diff
@@ -74,6 +99,60 @@ export async function sdkGeneratorApiHandler(
     };
 }
 
+function checkInvalidHttpMethods(
+    sdkClass: SdkClassConfiguration,
+    astClass: ClassDefinition,
+    language: Language,
+): { methods: MethodDefinition[]; methodsString: string } {
+    const methods: MethodDefinition[] = [];
+    let methodsString = "";
+    switch (language) {
+        case "ts":
+            for (const method of sdkClass.methods) {
+                if (method.type === TriggerType.http) {
+                    const astMethod = astClass.methods.find((m) => m.name === method.name);
+                    if (
+                        astMethod &&
+                        (astMethod.params.length != 1 ||
+                            astMethod.params[0].paramType.type != AstNodeType.RequestType ||
+                            astMethod.params[0].paramType.name != "Request" ||
+                            astMethod.params[0].optional != false ||
+                            astMethod.returnType.type != AstNodeType.PromiseType ||
+                            astMethod.returnType.generic.type != AstNodeType.ResponseType ||
+                            (astMethod.returnType.generic as ResponseType).name != "Response")
+                    ) {
+                        methods.push(astMethod);
+                        methodsString +=
+                            ` ${colors.red(`- ${astClass.name}.${astMethod.name}`)}` + "\n";
+                    }
+                }
+            }
+            return { methods, methodsString };
+        default:
+            methodsString = "";
+            for (const method of sdkClass.methods) {
+                if (method.type === TriggerType.http) {
+                    const astMethod = astClass.methods.find((m) => m.name === method.name);
+                    if (
+                        astMethod &&
+                        (astMethod.params.length != 1 ||
+                            astMethod.params[0].paramType.type != AstNodeType.RequestType ||
+                            astMethod.params[0].paramType.name != "Request" ||
+                            astMethod.params[0].optional != false ||
+                            astMethod.returnType.type != AstNodeType.PromiseType ||
+                            astMethod.returnType.generic.type != AstNodeType.ResponseType ||
+                            (astMethod.returnType.generic as ResponseType).name != "Response")
+                    ) {
+                        methods.push(astMethod);
+                        methodsString +=
+                            ` ${colors.red(`- ${astClass.name}.${astMethod.name}`)}` + "\n";
+                    }
```

Comment:
This is a lot of duplicated code here. This check should not be language-dependent, as we have the AST. Simply checking if the function has a single parameter of type `Request` and a return type of either `Response` or `Promise` with `generic = Response` should do the trick. There is no need to verify the names of the types here. The AST already did this